src/main/java/gyro/google/compute/RegionAutoscalerFinder.java

```
@@ -0,0 +1,86 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gyro.google.compute;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import com.google.api.services.compute.Compute;
+import com.google.api.services.compute.model.Autoscaler;
+import com.google.api.services.compute.model.RegionAutoscalerList;
+import com.psddev.dari.util.ObjectUtils;
+import com.psddev.dari.util.StringUtils;
+import gyro.core.GyroException;
+import gyro.core.Type;
+import gyro.google.GoogleFinder;
+import gyro.google.util.Utils;
+
+/**
+ * Query a region autoscaler.
+ *
+ * You can provide an expression that filters resources. The expression must specify the field name, and the value that you want to use for filtering.
+ *
+ * Please see :doc:`compute-region-autoscaler` resource for available fields.
+ *
+ * Example
+ * -------
+ *
+ * .. code-block:: gyro
+ *
+ *    region-autoscaler: $(external-query google::compute-region-autoscaler { name: 'compute-region-autoscaler-example', region: 'us-central1' })
+ */
+@Type("compute-region-autoscaler")
+public class RegionAutoscalerFinder extends GoogleFinder<Compute, Autoscaler, RegionAutoscalerResource> {
+
+    @Override
+    protected List<Autoscaler> findAllGoogle(Compute client) throws Exception {
+        return AutoscalerFinder.findAllAutoscalers(client, getProjectId(), ResourceScope.REGION, null);
+    }
+
+    @Override
+    protected List<Autoscaler> findGoogle(Compute client, Map<String, String> filters) throws Exception {
+        if (filters.containsKey("zone")) {
+            throw new GyroException("For zonal autoscaler, use 'compute-autoscaler' instead.");
+        }
+
+        String region = filters.remove("region");
+
+        if (StringUtils.isBlank(region) || ObjectUtils.isBlank(filters)) {
+            return AutoscalerFinder.findAllAutoscalers(client, getProjectId(), ResourceScope.REGION, filters);
+        }
```

The find specific finder should not return everything at any point. I would argue that it should throw an exception if region is blank.

---

it doesn't return everything, but still uses the filters a user provided, and what it returns is totally  up to the provider.